css-env-1/Overview.bs

```diff
@@ -77,6 +77,53 @@ tho user agents may define additional undocumented [=environment variables=]:
 
 * Issue: define list of predefined vars
 
+Safe area inset variables {#safe-area-insets}
+------------------------------------------------------------------
+
+<pre class=propdef>
+	Name: safe-area-inset-top
+	Applies to: all elements
+	Value: <<length>>
+	Initial: 0
+	Inherited: yes
+	Media: visual
+	Animatable: no
+</pre>
+
+<pre class=propdef>
+	Name: safe-area-inset-right
+	Applies to: all elements
+	Value: <<length>>
+	Initial: 0
+	Inherited: yes
+	Media: visual
+	Animatable: no
+</pre>
+
+<pre class=propdef>
+	Name: safe-area-inset-bottom
+	Applies to: all elements
+	Value: <<length>>
+	Initial: 0
+	Inherited: yes
+	Media: visual
+	Animatable: no
+</pre>
+
+<pre class=propdef>
+	Name: safe-area-inset-left
+	Applies to: all elements
+	Value: <<length>>
+	Initial: 0
+	Inherited: yes
+	Media: visual
+	Animatable: no
+</pre>
+
+The safe area insets are four [=environment variables=] that define the top, right, bottom and
+left insets from the edge of the viewport. Page content that is inside these insets may be
+obscured by the shape of the display.
```

Comment:
I think this description could be clearer about what these insets are for to readers not already familiar with the concept, and should probably also have stronger implementation conformance criteria.  I'd suggest rewording to something like the following:

> The safe area insets are four [=environment variables=] that define a rectangle by its top, right, bottom, and left insets from the edge of the viewport.  For rectangular displays, these must all be zero, but for nonrectangular displays they must form a rectangle, chosen by the user agent, such that all content inside the rectangle is visible, and such that reducing any of the insets would cause some content inside of the rectangle to be invisible due to the nonrectangular nature of the display.  This allows authors to limit the layout of essential content to the space inside of the safe area rectangle.

Reply:
Fixed